### PR TITLE
in-progress PayGroup implementation (early submission to give Pierre what he needs)

### PR DESCRIPTION
**Pull request by taoeffect**
Submitting this for an early merge to get @pieer the `groupIncomeDistribution` getter. Includes misc. fixes + in-progress PayGroup page

### CODE REVIEW

File: frontend/model/state.js [hunk 1]
```diff
@@ -95,6 +98,13 @@ sbp('sbp/selectors/register', {
 // http://vuex.vuejs.org/en/mutations.html
 const mutations = {
   login (state, user) {
+    const monthTimestamp = currentMonthTimestamp()
```
Comment (pieer) [inline frontend/model/state.js:103]: Could you had a comment of why you initialise the paymentsByMonth on login?

File: frontend/model/state.js [hunk 2]
```diff
@@ -250,6 +263,25 @@ const getters = {
     const currency = currencies[getters.groupSettings.mincomeCurrency]
     return currency && currency.symbolWithCode
   },
+  groupIncomeDistribution (state, getters) {
+    const profiles = getters.groupMembers
+    const mincomeAmount = getters.groupMincomeAmount
+    const currentIncomeDistribution = []
+    for (const username in profiles) {
+      const profile = profiles[username]
+      const incomeDetailsType = profile && profile.groupProfile.incomeDetailsType
```
Comment (pieer) [inline frontend/model/state.js]: should you use !!profile.groupProfile.incomeDetailsType instead?

File: frontend/model/state.js (hunk 1 above)
Comment (taoeffect) [reply frontend/model/state.js:103]: done (when next commit is pushed)

File: frontend/model/state.js (hunk 2 above)
Comment (pieer) [inline frontend/model/state.js]: Could use !!profile.groupProfile.incomeDetailsType

Comment (taoeffect) [reply frontend/model/state.js]: No since that would return a boolean, and I need its value to retrieve the income/pledgeAmount

**Review by pieer — APPROVED** [review]

**Comment by taoeffect** [comment]
**NOTE: this is an exceptional merge that violates Rule 7 of our contributions guidelines because it's during a hackathon and I have approval from @pieer to merge it, and @pierre cannot merge it himself because the build is failing, but it's an error unrelated to this PR.**

**Pull request merged**